simtools/simulator.py

```diff
@@ -0,0 +1,740 @@
+            Single file or list of files of shower simulations.
+
+        Returns
+        -------
+        runs_and_files: dict
+            dictionary with runnumber as key and (if availble) simulation
+            file name as value
+
+        """
+
+        _runs_and_files = {}
+
+        if self.simulator == 'simtel':
+            _file_list = self._enforceListType(inputFileList)
+            for file in _file_list:
+                _runs_and_files[self._guessRunFromFile(file)] = file
+        else:
+            _run_list = self._getRunsToSimulate()
+            for run in _run_list:
+                _runs_and_files[run] = None
+
+        return _runs_and_files
+
+    @staticmethod
+    def _enforceListType(inputFileList):
+        """Enforce the input list to be a list."""
+        if not inputFileList:
+            return list()
+        elif not isinstance(inputFileList, list):
+            return [inputFileList]
+
+        return inputFileList
+
+    def _guessRunFromFile(self, file):
+        """
+        Finds the run number for a given input file name.
+        Input file names can follow any pattern with the
+        string 'run' followed by the run number.
+        If not found, returns 1.
+
+        Parameters
+        ----------
+        file: Path
+            Simulation file name
+
+        """
+        fileName = str(Path(file).name)
+
+        try:
+            runStr = re.search("run[0-9]*", fileName).group()
+            return int(runStr[3:])
+        except (ValueError, AttributeError):
+            msg = "Run number could not be guessed from {} using run = 1".format(fileName)
+            self._logger.warning(msg)
+            return 1
+
+    def _fillResults(self, file, run):
+        """
+        Fill the results dict with input, output and log files.
+
+        Parameters
+        ----------
+        file: str
+            input file name
+        run: int
+            run number
+
+        """
+
+        self._results["output"].append(str(self._simulationRunner.getOutputFile(run)))
+        self._results["sub_out"].append(str(self._simulationRunner.getSubLogFile(run, mode="out")))
+        self._results["log"].append(str(self._simulationRunner.getLogFile(run)))
+        if self.simulator == 'simtel':
+            self._results["input"].append(str(file))
+            self._results["hist"].append(str(self._simulationRunner.getHistogramFile(run)))
+        else:
+            self._results["input"].append(None)
+            self._results["hist"].append(None)
+
+    def printHistograms(self, inputFileList=None):
+        """
+        Print histograms and save a pdf file.
+
+        Parameters
+        ----------
+        inputFileList: str or list of str
+            Single file or list of files of shower simulations.
+
+        Returns
+        -------
+        path
+            Path of the pdf file.
+        """
+
+        figName = None
+
+        if self.simulator == 'simtel':
```

Comment:
Should we print a warning if this function is called for a different simulator? Will help avoiding mistakes when writing applications for example.

Reply:
OK!